core/src/main/java/uk/gov/gchq/koryphe/iterable/ChainedIterator.java

```diff
@@ -63,12 +65,16 @@ private Iterator<? extends T> getIterator() {
         while (!currentIterator.hasNext()) {
             CloseableUtil.close(currentIterator);
             if (iterablesIterator.hasNext()) {
-                currentIterator = iterablesIterator.next().iterator();
+                Object next = iterablesIterator.next();
+                if (nonNull(next) && next instanceof Iterable) {
+                    currentIterator = (Iterator<? extends T>) ((Iterable<?>) next).iterator();
+                } else if (nonNull(next) && !(next instanceof Iterable)) {
+                    throw new IllegalStateException(String.format("Iterator of Iterator contains non-iterable class: %s object: %s", next.getClass(), next));
```

Comment:
Should it be `Iterator of Iterables`?

Reply:
The Function required is `KorypheFunction<Iterable<Iterable<I_ITEM>>, Iterable<I_ITEM>>` 
so neither ? :D

Reply:
```suggestion
                    throw new IllegalStateException(String.format("Iterable of Iterable contains non-iterable class: %s object: %s", next.getClass(), next));
```